lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,137 @@
+  std::int16_t acceleration
+    = static_cast<std::int16_t>((static_cast<int32_t>(rawAcceleration) * 488) / 1000);
+  return acceleration;
+}
+
+// TODOlater: current settings of the accelerometer make it read in +-16g but with high noise. Check
+// whether other configuration could be better
+std::optional<core::RawAccelerationData> Accelerometer::read()
+{
+  // check to see if the values are ready to be read
+  const auto data_ready = i2c_.readByte(kDeviceAddress, kDataReady);
+  if (!data_ready) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read acceleration data");
+    return std::nullopt;
+  }
+  // TODOlater: here the error is not that bad, hence the return value should indicate that
+  if (data_ready.value() % 2 == 0) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read acceleration data as it is not ready");
+    return std::nullopt;
+  }
+  const auto result_x = getRawAcceleration(Axis::x);
+  if (!result_x) { return std::nullopt; }
+  const std::int16_t x_acceleration = getAccelerationFromRaw(result_x.value());
+  const auto result_y               = getRawAcceleration(Axis::y);
+  if (!result_y) { return std::nullopt; }
+  const std::int16_t y_acceleration = getAccelerationFromRaw(result_y.value());
+  const auto result_z               = getRawAcceleration(Axis::z);
+  if (!result_z) { return std::nullopt; }
+  const std::int16_t z_acceleration = getAccelerationFromRaw(result_z.value());
+  const std::optional<core::RawAccelerationData> acceleration_3_axis{
+    std::in_place,
+    x_acceleration,
+    y_acceleration,
+    z_acceleration,
+    std::chrono::system_clock::now()};
+  logger_.log(core::LogLevel::kDebug, "Successfully read accelerometer data");
+  return acceleration_3_axis;
+}
+
+core::Result Accelerometer::configure()
+{
+  // check we are communicating with the correct sensor
+  const auto device_id = i2c_.readByte(kDeviceAddress, kDeviceId);
+  if (!device_id) {
+    logger_.log(core::LogLevel::kFatal, "Failure to read device id of accelerometer");
+    return core::Result::kFailure;
+  }
+  if (device_id.value() != kExpectedDeviceId) {
+    logger_.log(core::LogLevel::kFatal, "Failure accelerometer didn't give correct device id");
+    return core::Result::kFailure;
+  }
+  // Sampling rate of 200 Hz
+  // Enable high performance mode
+  const core::Result ctrl1_result
+    = i2c_.writeByteToRegister(kDeviceAddress, kCtrl1Address, kCtrl1Value);
+  if (ctrl1_result == core::Result::kFailure) { return core::Result::kFailure; };
+  // Enable block data update
+  // Enable address auto increment
+  const core::Result ctrl2_result
+    = i2c_.writeByteToRegister(kDeviceAddress, kCtrl2Address, kCtrl2Value);
+  if (ctrl2_result == core::Result::kFailure) { return core::Result::kFailure; };
+  // Full scale +-16g
+  // Filter bandwidth = ODR/2
+  const core::Result ctrl6_result
+    = i2c_.writeByteToRegister(kDeviceAddress, kCtrl6Address, kCtrl6Value);
```

Comment:
should these comments be here or with the values in the header? I'm not sure which is better